### DISCUSSION TITLE
demographic vaccine data: % of vaccinations vs % of population

### DISCUSSION THREAD

**Discussion by smcclure17** [covid-projections/can-scrapers#270] [Scraper help]
The [KFF vaccine demographic data](https://www.kff.org/other/state-indicator/covid-19-vaccinations-by-race-ethnicity/?currentTimeframe=0&sortModel=%7B%22colId%22:%22Location%22,%22sort%22:%22asc%22%7D) appears to return data as percentages of the total # of vaccinations, for each race or ethnicity. The about section says: "_Percentages are based on total vaccinations for which race/ethnicity is known_." I believe meaning:  
`(# of vaccinations for demographic group) / (# of total vaccinations)`.

If I'm not mistaken, we have been collecting demographic data as percentages of total demographic population, or 
`(# of vaccinations for demographic group) / (demographic group population)`. 

Do we have a method of collecting percentages w/ this different denominator? Is there a way to make this compatible with our current schema? Or, should we add a new variable to the schema?

I've [collected the data](https://github.com/covid-projections/can-scrapers/compare/kff_vaccine_demographics?expand=1), but am not sure where to go from here.

**Comment by sglyon** [comment]
Hmm... from what you wrote and what I could see on the source site I think this is a problem. 

We don't have any other time series where we are reporting `(# of vaccinations for demographic group) / (# of total vaccinations)`

We should have both the `(# of total vaccinations)` from the CDC (state level) as well as `(demographic group population)` from the census.  I think we could in theory get the variable we want by computing `x * y / z` where:

- `x = (# of vaccinations for demographic group) / (# of total vaccinations)` (what KKF reports)
- `y = (# of total vaccinations)` (from CDC state vaccine report)
- `z = (demographic group population)` (from census)

This may belong somewhere else in the pipeline though as it goes beyond "get the raw data"


What do you think @mikelehen  and @TomGoBravo

**Reply by mikelehen** [reply]
Gah, yeah, that's a good flag.  @BrettBoval Do we know if Surgo wants this data (state-level vaccine demographics)?  It sounds like the "right" approach is:
1. Update the CAN scraper schema to be able to represent this.
2. Augment the CAN pipeline to combine KFF + CDC + Census data to "back-compute" the data that fits our API schema.

Given this is nontrivial additional scope, my gut is we should just drop scraping the KFF data for now given it was a "nice-to-have" that seemed easy at a glance.

**Reply by BrettBoval** [reply]
Agreed. I'm okay with this not getting scraped at the moment. Surgo is interested in the county level data primarily.

**Reply by TomGoBravo** [reply]
I imagine people at KFF are spending time maintaining this dataset. Has anybody from CAN spoken with them about making it more useful... at least for us?

**Reply by mikelehen** [reply]
I don't think we have any contact with them.  I'll defer to @BrettBoval on whether it's worth reaching out and how to coordinate.
